### PR TITLE
Do not ignore space_before_bound in where clause

### PR DESCRIPTION
**Pull request by stepancheg**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -358,9 +369,9 @@ impl Rewrite for ast::WherePredicate {
                                                     .collect());
 
                     if context.config.spaces_within_angle_brackets && lifetime_str.len() > 0 {
```
Comment (nrc) [inline src/types.rs]: Above, when calculating `used_width`, I think you need to take into account the width of `colon`, rather than a hard-coded number,

File: tests/source/space-before-bound.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // rustfmt-space_before_bound: true
 
 trait Trait {}
-fn f<'a, 'b: 'a, T: Trait>() {}
+trait Trait2 {}
+fn f<'a, 'b: 'a, T: Trait, U>() where U: Trait2 {}
```
Comment (nrc) [inline tests/source/space-before-bound.rs]: Could you add a test where you hit the max width and overflow please?

Comment (stepancheg) [inline tests/source/space-before-bound.rs]: `used_width` fixed, test added. Updated the PR.

**Comment by nrc** [comment]
Thanks!
